### PR TITLE
Fix detection of preview versions

### PR DESCRIPTION
**Pull request by theunrepentantgeek**
With the change to include the Generator version `v1alpha1` in the generated packages, **_all_** versions were being identified as preview.

This was incorrect - the `IsPreview()` test should ignore the version of the generator and pay attention only to the API version.

### CODE REVIEW

**Review by Porges — APPROVED** [review]

File: hack/generator/pkg/astmodel/local_package_reference.go [hunk 1]
```diff
@@ -20,9 +21,32 @@ type LocalPackageReference struct {
...
 func MakeLocalPackageReference(prefix string, group string, version string) LocalPackageReference {
-	return LocalPackageReference{localPathPrefix: prefix, group: group, version: version}
+	return LocalPackageReference{
+		localPathPrefix: prefix,
+		group:           group,
+		version:         version,
+	}
+}
+
+func CreateLocalPackageNameFromVersion(version string) string {
```
Comment (matthchr) [inline hack/generator/pkg/astmodel/local_package_reference.go:35]: minor: Missing comment

**Review by matthchr — APPROVED** [review]

File: hack/generator/pkg/astmodel/package_reference.go [hunk 2]
```diff
@@ -269,8 +269,9 @@ func (v *versionComparer) isPreviewVersionLabel(identifier string) (int, bool) {
 // special set, and if so returns its true. If the passed identifier does not contain one,
 // returns false.
 func containsPreviewVersionLabel(identifier string) bool {
+
```
Comment (babbageclunk) [inline hack/generator/pkg/astmodel/package_reference.go:272]: Rogue newline?

**Review by babbageclunk — APPROVED** [review]

File: hack/generator/pkg/astmodel/local_package_reference.go (hunk 1 above)
Comment (theunrepentantgeek) [reply hack/generator/pkg/astmodel/local_package_reference.go:35]: Fixed.

File: hack/generator/pkg/astmodel/package_reference.go (hunk 2 above)
Comment (theunrepentantgeek) [reply hack/generator/pkg/astmodel/package_reference.go:272]: Fixed.

**Pull request merged**